cmd/kubectl-gadget/seccompadvisor.go

```diff
@@ -0,0 +1,179 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+var seccompAdvisorCmd = &cobra.Command{
+	Use:   "seccomp-advisor",
+	Short: "Generate seccomp policies based on recorded syscalls activity",
+}
+
+var seccompAdvisorStartCmd = &cobra.Command{
+	Use:          "start",
+	Short:        "Start to monitor the system calls",
+	RunE:         runSeccompAdvisorStart,
+	SilenceUsage: true,
+}
+
+var seccompAdvisorStopCmd = &cobra.Command{
+	Use:          "stop",
+	Short:        "Stop monitoring and report the policies",
+	RunE:         runSeccompAdvisorStop,
+	SilenceUsage: true,
+}
+
+var seccompAdvisorListCmd = &cobra.Command{
+	Use:          "list",
+	Short:        "List existing seccomp traces",
+	RunE:         runSeccompAdvisorList,
+	SilenceUsage: true,
+}
+
+var (
+	outputMode         string
+	seccompProfileName string
+)
+
+func init() {
+	// Add generic information.
+	rootCmd.AddCommand(seccompAdvisorCmd)
+	utils.AddCommonFlags(seccompAdvisorCmd, &params)
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorStartCmd)
+	seccompAdvisorStartCmd.PersistentFlags().StringVarP(&outputMode, "output-mode", "m", "terminal", "The trace output mode, possibles values are ExternalResource or Status (default).")
+	seccompAdvisorStartCmd.PersistentFlags().StringVar(&seccompProfileName, "seccomp-profile-name", "", "The seccomp profile which will be created when using ExternalResource as value for --output-mode.")
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorStopCmd)
+	seccompAdvisorCmd.AddCommand(seccompAdvisorListCmd)
+}
+
+func outputModeToTraceOutputMode(outputMode string) (string, error) {
+	switch outputMode {
+	case "terminal":
+		return "Status", nil
+	case "seccomp-profile":
+		return "ExternalResource", nil
+	default:
+		return "", fmt.Errorf("%q is not an accepted value for --output-mode, possible values are: terminal (default) and seccomp-profile", outputMode)
+	}
+}
+
+// runSeccompAdvisorStart starts monitoring of syscalls for the given
+// parameters.
+func runSeccompAdvisorStart(cmd *cobra.Command, args []string) error {
+	if params.Podname == "" {
+		return errors.New("Usage: kubectl gadget seccompadvisor start -p podname")
+	}
+
+	traceOutputMode, err := outputModeToTraceOutputMode(outputMode)
+	if err != nil {
+		return err
+	}
+
+	if traceOutputMode != "ExternalResource" && seccompProfileName != "" {
+		return errors.New("Can only use --seccomp-profile-name with --output seccomp-profile.")
+	}
```

Comment:
Please also validate that `--seccomp-profile-name` is passed when `--output seccomp-profile` is used.